src/pthread-main.js

```diff
@@ -0,0 +1,96 @@
+// Pthread Web Worker startup routine:
+// This is the entry point file that is loaded first by each Web Worker
+// that executes pthreads on the Emscripten application.
+
+// All pthreads share the same Emscripten HEAP as SharedArrayBuffer
+// with the main execution thread.
+var buffer;
+
+var threadBlock = 0; // Info area for this thread in Emscripten HEAP (shared). If zero, this worker is not currently hosting an executing pthread.
```

Comment:
This points to the malloced `pthread_t` structure, of which there exists one for each thread. This pointer is also used to denote the ID of the thread. I liked using the term block since from JS perspective it's a pointer to a block of C memory structure, and that term is not a commonly used term elsewhere in the code, so it seemed to give a unique meaning. I can rename it though, perhaps `threadStruct`? or `threadMemoryBlock`? or `threadInfoStruct`?